addons/calendar/models/calendar_recurrence.py

```diff
@@ -361,6 +361,26 @@ def _rrule_parse(self, rule_str, date_start):
         data = {}
         day_list = ['mon', 'tue', 'wed', 'thu', 'fri', 'sat', 'sun']
 
+        # Skip X-named RRULE extensions
+        # TODO Remove patch when dateutils contains the fix
+        # HACK https://github.com/dateutil/dateutil/pull/1374
+        try:
+            prefix, suffix = rule_str.split(':', 1)
+        except ValueError:
+            # No special params, that's OK
+            pass
+        else:
+            new_prefix = ";".join(
+                part for part in prefix.split(';') if not part.startswith('X-')
+            )
+            if new_prefix:
+                # If there are non-X-named params, keep only those. FWIW it
+                # will mean an ValueError in dateutil
+                rule_str = f"{new_prefix}:{suffix}"
+            else:
+                # If all RRULE params were X-extensions, just keep the suffix
+                rule_str = suffix
+
```

Comment:
I made a review commit with a proposal that could probably be improved. I also updated your testing case to put several X- parameter into a single RRULE. Once again, the test would be better inside the calendar module but that's a start. :-)

Reply:
Thanks, I didn't see the case you mention in my logs so I didn't fix it, but indeed it seems possible too.

Regarding the position of the colon `:`, check out [RFC 5545 3.8.5.3](https://datatracker.ietf.org/doc/html/rfc5545#section-3.8.5.3):

>    Format Definition:  This property is defined by the following notation:
> 
>        rrule      = "RRULE" rrulparam ":" recur CRLF
> 
>        rrulparam  = *(";" other-param)

So, it seems [Evolution uses the `rruleparam` part to add its customization](https://discourse.gnome.org/t/working-with-evolution-mail-and-web-calendar-using-rrule-fail-because-of-x-evolution-enddate-in-rrule/19710).

Just trying to explain the motivation, but I agree with all you said.